presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsCalculator.java

```diff
@@ -122,22 +118,21 @@ public VariableStatsEstimate visitCall(CallExpression call, Void context)
                 return computeArithmeticBinaryStatistics(call, context);
             }
 
-            Object value = new RowExpressionInterpreter(call, metadata, session.toConnectorSession(), OPTIMIZED).optimize();
+            RowExpression value = new RowExpressionOptimizer(metadata).optimize(call, OPTIMIZED, session.toConnectorSession());
 
             if (value == null) {
                 return nullStatsEstimate();
             }
 
-            if (value instanceof RowExpression) {
-                // value is not a constant
-                return VariableStatsEstimate.unknown();
+            if (value instanceof ConstantExpression) {
+                return value.accept(this, context);
```

Comment:
the difference between visitConstant and what visitCall used to do with constants is that visitConstant will also  set the min and max values for numeric constants. As far as I can tell, that seems like something you want here too, so this would be a good change.

Reply:
Thanks @rschlussel  for the comment. I will keep it then.